### PR TITLE
Use SavedState class for local storage

### PR DESCRIPTION
**Pull request by NoSloppy**
How about this?

### CODE REVIEW

File: style_editor.html [hunk 1]
```diff
@@ -10018,16 +10077,117 @@
...
+      }
+    });
+  }
+}
+
+// Call the onPageLoad function when the page is loaded
+window.addEventListener('DOMContentLoaded', onPageLoad);
+
+function onPageLoad() {
+  inhiltState.onload();
```
Comment (profezzorn) [inline style_editor.html:10140]: I think this should be a loop over all_saved_states

File: style_editor.html [hunk 2]
```diff
@@ -10018,16 +10077,117 @@
...
+
+/* Settings buttons saved as local storage */
+function getSavedState(buttonState, defaultValue) {
+  var value = localStorage.getItem(buttonState);
+  console.log("Retrieved SavedState for " + buttonState + ": " + value);
+
+  if (value === null) {
+    return defaultValue;
+  }
+  // Convert the string to boolean unless it's null or "false"
```
Comment (profezzorn) [inline style_editor.html:10152]: It can't be null when we get here...

File: style_editor.html [hunk 3]
```diff
@@ -10018,16 +10077,117 @@
...
             onclick="ClickRotate()">
             <i class="fa-solid fa-arrows-rotate"></i> Rotate</button>
-          <input
-            type=button id=INHILT_BUTTON name=inhilt  value="Inhilt"
-            title="Simulates In-Hilt LED blades, (no addressable pixel effects)."
-            onclick="ClickInhilt()" >
-          <input
-            type=button id=SLOW_BUTTON name=inhilt value="Slow"
-            title="Provides &quot;snapshots&quot; of rapidly moving things to see what they look like without fast movement. *Note* Graflex hilt model probably won't be able to render with this on."
-            onclick="ClickSlow()" >
```
Comment (profezzorn) [inline style_editor.html:10028]: We can get rid of ClickSlow and ClickInhilt, right?

File: style_editor.html [hunk 4]
```diff
@@ -10018,16 +10077,117 @@
...
+  }
+  get() { return this.value; }
+}
+
+var inhiltState = new SavedState("inhilt", false, (on) => { STATE_NUM_LEDS = on ? 1 : 144; });
+var slowState = new SavedState("slow", false, (on) => { framesPerUpdate = on ? 60 : 0; });
+
+function handleClick(checkbox) {
+  var stateName = checkbox.dataset.state;
+  var state = window[stateName];
```
Comment (profezzorn) [inline style_editor.html:10186]: I'm not a big fan of looking things up in "window"
can we have an explicit map from checkbox to state so we can do:

    var state = state_by_checkbox[checkbox];

instead?
We should be able to add the state to the state_by_checkbox dict from inside the SavedState constructor.

File: style_editor.html (hunk 3 above)
Comment (NoSloppy) [reply style_editor.html:10028]: Yes they are gone. the buttons are in settings panel and all settings checkboxes and saves can be handled with handleClick().

File: style_editor.html (hunk 2 above)
Comment (NoSloppy) [reply style_editor.html]: old comment from before the above line. deleted.

File: style_editor.html (hunk 1 above)
Comment (NoSloppy) [reply style_editor.html]: Good idea, considering there are more buttons to come.

File: style_editor.html (hunk 3 above)
Comment (profezzorn) [reply style_editor.html:10028]: I don't see any delete lines for those functions though.

Comment (NoSloppy) [reply style_editor.html:10028]: My mistake. deleted now.

**Pull request merged**